File: src/main/java/com/indiepost/model/PostReading.java
package com.indiepost.model;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * Created by jake on 7/25/16.
 */
@Entity
@Table(name = "PostReadings")
public class PostReading implements Serializable {

    private static final long serialVersionUID = 5793341286387473902L;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "userId", nullable = false)
    private User user;

    @Column(name = "userId", insertable = false, updatable = false)
    private Long userId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "postId", nullable = false)
    private Post post;

    @Column(name = "postId", insertable = false, updatable = false)
    private Long postId;

    @Column(nullable = false)
    private LocalDateTime created;

    @Column(nullable = false)
    private LocalDateTime lastRead;

    @Column(nullable = false)
    private int readCount = 1;

    private boolean visible = true;

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public Long getPostId() {
        return postId;
    }

    public void setPostId(Long postId) {
        this.postId = postId;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Post getPost() {
        return post;
    }

    public void setPost(Post post) {
        this.post = post;
    }

    public LocalDateTime getLastRead() {
        return lastRead;
    }

    public void setLastRead(LocalDateTime lastRead) {
        this.lastRead = lastRead;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public boolean isVisible() {
        return visible;
    }

    public void setVisible(boolean visible) {
        this.visible = visible;
    }

    public LocalDateTime getCreated() {
        return created;
    }

    public void setCreated(LocalDateTime created) {
        this.created = created;
    }

    public int getReadCount() {
        return readCount;
    }

    public void setReadCount(int readCount) {
        this.readCount = readCount;
    }

    public void increaseReadCount() {
        this.readCount++;
    }
}

File: src/main/kotlin/com/indiepost/model/PostReading.kt
package com.indiepost.model

import java.time.LocalDateTime
import javax.persistence.*

/**
 * Created by jake on 7/25/16.
 */
@Entity
@Table(name = "Posts_Users_Readings")
data class PostReading(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        var id: Long? = null,

        @Column(name = "userId", insertable = false, updatable = false)
        var userId: Long? = null,

        @Column(name = "postId", insertable = false, updatable = false)
        var postId: Long? = null,

        @Column(nullable = false)
        var created: LocalDateTime? = null,

        @Column(nullable = false)
        var lastRead: LocalDateTime? = null,

        @Column(nullable = false)
        var readCount: Int = 1,

        var isVisible: Boolean = true,

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "userId", nullable = false)
        var user: User? = null,

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "postId", nullable = false)
        var post: Post? = null
) {
    fun increaseReadCount() {
        this.readCount = readCount + 1
    }
}